imap_processing/swe/l1b/swe_l1b.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from imap_processing import imap_module_directory
+from imap_processing.swe.l1a.swe_l1a import SWEAPID
+from imap_processing.swe.l1b.swe_l1b_science import swe_l1b_science
+
+
+def convert_raw_to_eu(dataset: xr.Dataset):
+    """Convert raw data to engineering unit.
+
+    Parameters
+    ----------
+    dataset : xr.Dataset
+
+    Returns
+    -------
+    xr.Dataset
+        Raw data converted to engineering unit as needed.
+    """
+    test_data_path = f"{imap_module_directory}/swe/l1b/"
+
+    eu_conversion_table = pd.read_csv(
+        f"{test_data_path}/engineering_unit_convert_table.csv",
+        index_col="index",
+    )
+
+    # Look up packet name from APID
+    packet_name = next(
+        packet for packet in SWEAPID if packet.value == dataset["PKT_APID"].data[0]
+    )
+
+    # Look up all metadata fields for the packet name
+    metadata_list = eu_conversion_table.loc[
+        eu_conversion_table["packetName"] == packet_name.name
+    ]
+
+    # for each metadata field, convert raw value to engineering unit
+    for field in metadata_list.index:
+        metadata_field = metadata_list.loc[field]["mnemonic"]
+        # On this line, we are getting the coefficients from the
+        # table and then reverse them because the np.polyval is
+        # expecting coefficient in decending order
+        coeff_values = metadata_list.loc[
+            metadata_list["mnemonic"] == metadata_field
+        ].values[0][6:][::-1]
+
+        # Convert the raw value to engineering unit
+        dataset[metadata_field].data = np.polyval(
+            coeff_values, dataset[metadata_field].data
+        )
```

Comment:
Use the `Polynomial` class instead because polyval is discouraged. It looks like it is also in increasing order for the coefficients.

https://numpy.org/doc/stable/reference/generated/numpy.polynomial.polynomial.Polynomial.html#numpy.polynomial.polynomial.Polynomial